state/indexer/block/kv/kv.go

```diff
@@ -601,9 +620,32 @@ func (idx *BlockerIndexer) indexEvents(batch dbm.Batch, events []abci.Event, hei
 				if err := batch.Set(key, heightBz); err != nil {
 					return err
 				}
+				keyArray = AppendToKeyArray(keyArray, key)
 			}
 		}
 	}
+	err := batch.Set(getEventsForHeightKey(height), keyArray)
+	return err
+}
 
-	return nil
+func (idx *BlockerIndexer) deleteEvents(height int64) {
+	eventsKey := getEventsForHeightKey(height)
+	keyArray, err := idx.store.Get(eventsKey)
+	if err != nil {
```

Comment:
Maybe here it's worth checking whether the error is `ErrNotFound` and not panicking.